File: easy/ts/FuncCall.ts
import { VulnChild } from './common/VulnChild';
import { InterfaceImpl } from './cmd_exec/InterfaceImpl';
import { VulnRunner } from './iface/VulnRunner';

export class FuncCall {
    static runByChild(cmd: string): void {
        const child = new VulnChild();
        child.exec(cmd);
    }

    static runByInterface(cmd: string): void {
        const runner: VulnRunner = new InterfaceImpl();
        runner.run(cmd);
    }
}

File: easy/dart/func_call.dart
import 'common/vuln_child.dart';
import 'cmd_exec/interface_impl.dart';
import 'iface/vuln_runner.dart';

class FuncCall {
  static void runByChild(String cmd) {
    VulnChild child = VulnChild();
    child.exec(cmd);
  }

  static void runByInterface(String cmd) {
    VulnRunner runner = InterfaceImpl();
    runner.run(cmd);
  }
}
